File: blogServer/models/Blog.js
import mongoose from "mongoose";

const BlogSchema = new mongoose.Schema({
  owner: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "User", 
    required: true,
  },
  title: {
    type: String,
    required: true,
  },
  content: {
    type: String,
    required: true,
  },
  image: {
    type: String,
    default: "",
  },
  tags: {
    type: [String],
    default: [],
  },
  keywords: {
    type: [String],
    default: [],
  },
  likes: {
    type: Number,
    default: 0,
  },
  dislikes:{
    type: Number,
    default: 0,
  }
}, { timestamps: true });

export const  Blog = mongoose.model("Blog", BlogSchema,"Blogs");

File: tagService/models/blog.py
from pydantic import BaseModel, Field
from typing import List, Optional
from bson import ObjectId
import motor.motor_asyncio
import os
from dotenv import load_dotenv

load_dotenv()

# Helper to handle ObjectId serialization
class PyObjectId(ObjectId):
    @classmethod
    def __get_validators__(cls):
        yield cls.validate

    @classmethod
    def validate(cls, v):
        if not ObjectId.is_valid(v):
            raise ValueError("Invalid ObjectId")
        return ObjectId(v)

# Pydantic model for validation
class BlogModel(BaseModel):
    id: Optional[PyObjectId] = Field(alias="_id")
    owner: PyObjectId
    title: str
    content: str
    image: Optional[str] = ""
    tags: List[str] = []
    keywords: List[str] = []
    likes: int = 0
    dislikes: int = 0

    class Config:
        allow_population_by_field_name = True
        arbitrary_types_allowed = True
        json_encoders = {ObjectId: str}

# MongoDB setup
MONGO_URI = os.getenv("MONGODBURL")+"/"+os.getenv("MONGODBDBNAME")

client = motor.motor_asyncio.AsyncIOMotorClient(MONGO_URI)
db = client["blogs"]
blogs_collection = db["Blogs"]
